### PR TITLE
expose init expression on var-declaration parser hook

### PR DESCRIPTION
**Pull request by id0Sch**
**What kind of change does this PR introduce?**
PR for https://github.com/webpack/webpack/issues/7636

**Did you add tests for your changes?**
yes, Parser.js tests, I hope that I made them in the right place.

**Does this PR introduce a breaking change?**
Adds a second argument to the varDecalaration parser hook

**What needs to be documented once your changes are merged?**
The new arguemnt (`initExpression`)

### CODE REVIEW

**Comment by id0Sch** [comment]
@sokra How can I handle the patch/integration failure?

File: test/Parser.unittest.js [hunk 1]
```diff
@@ -246,6 +246,50 @@ describe("Parser", () => {
...
+			{
+				smu: "ArrayExpression"
+			}
+		],
+		"array destruct var decalaration": [
+			function() {
+				var [{ smu }] = [{ testVar: "abc" }, { otherVar: "bcd" }];
+			},
+			{
+				smu: "ArrayExpression"
```
Comment (sokra) [inline test/Parser.unittest.js]: This is not correct. It should be null or undefined.

Comment (id0Sch) [reply test/Parser.unittest.js]: that's what came out of `declarator.init`, could you please elaborate why so I can try and figure out what went wrong?

Sorry about the delay, I've been caught up in so many things this summer.

Comment (sokra) [reply test/Parser.unittest.js]: `var [{ smu }] = [{ testVar: "abc" }, { otherVar: "bcd" }];`

This destructs the array and object resulting in `smu` being undefined and not an `ArrayExpression`. You need to do the same destruction when determining the initExpresssion for a variable.

**Pull request closed without merging**